docs/src/single-validator-manager.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Single Validator Stake Pool Manager Program
+---
+
+A work-in-progress program for permissionless management of stake pools that are
+only delegated to one validator.
+
+## Overview
+
+The single-validator manager program defines a canonical stake pool for any
+validator on the network, which can be created by anyone.
+
+The program manages the "delegation strategy" of the pool, which is to maximize
+the amount of stake on one validator, while maintaining a small amount of SOL
+in the reserve for small stakers.
+
+## Fees
+
+Pools are configured with a SOL deposit fee of 8 basis points, which corresponds
+to the rewards received over two epochs. This way, no user can deposit SOL and
+immediately withdraw a stake account at the end of an epoch to steal rewards
+from the pool.
+
+All other fees are 0 so that pools mimic the reward structure of native staking.
+
+The program also exposes a permissionless `BurnFees` instruction to allow anyone
+to burn the pool tokens in the fee account, which increases the scarcity of all
+other pool tokens.
+
+## Operation
+
+### Create Pool
+
+The `CreateStakePool` instruction allows anyone to create the canonical stake
+pool for a validator, at the cost of ~0.0124 SOL for account rent-exemption of
+the stake pool, validator list, reserve, mint, and fee account.
+
+After the pool is created, the reserve requires at least stake rent-exemption plus
+the minimum delegation amount to actually add the validator to the pool, which
+costs 1.0028288 SOL, assuming a minimum delegation of 1 SOL. The user receives
+pool tokens for this deposit.
+
+After the SOL deposit, the `AddValidatorToPool` instruction on the pool management
+program allows anyone to add the stake account for that validator.
+
+Once the added stake account is active, anyone may deposit stake accounts using
+the `DepositStake` instruction on the stake pool, and not using the manager program.
+
+### Depositing and withdrawing stake accounts
+
+If stakers deposit or withdraw stake accounts from the pool, everything works
+seamlessly. Stakers receive pool tokens for their deposited stake account, and
+may use the pool tokens to withdraw a stake account.
+
+Activated stake accounts on the same validator are all fungible, so there's no
+other economics to worry about.
+
+### Depositing and withdrawing SOL
+
+As mentioned in the fees section, only SOL deposits have a fee, and everything
+else can be done freely.
+
+To maximize returns for pool depositors, the program allows anyone to maximally
+deploy the reserve to the validator, with important limits.
+
+Single-validator stake pools are an option for small stakers to earn inflation
```

Comment:
@joncinque / @brooksprumo - it really seems like if `get_minimum_delegation()` was changed to only affect epoch rewards, then all this small-stake complexity goes away as a small stake holder could withdraw their stake the same as the larger holders.

Reply:
Hoo boy, yes it would. We could always do this in two passes: the first version has this annoyance, and then if small stakes are eventually excluded from rewards, we can update the program to handle that.

Reply:
Yeah I kinda think so. 😬 
What kind of churn would that cause?

Reply:
It's more like antichurn, since it would involve reverting lots of changes :sweat_smile: